inference-engine/src/vpu/myriad_plugin/myriad_plugin.cpp

```diff
@@ -98,24 +135,128 @@ void Engine::QueryNetwork(
         VPU_THROW_UNLESS(!(std::find(deviceIDs.begin(), deviceIDs.end(), deviceName) == deviceIDs.end()), "Myriad device: {} not found.", deviceName);
     }
 
-    if (network.getFunction()) {
-        THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str << " ngraph::Function is not supported natively";
-    }
+    auto function = network.getFunction();
+
+    if (function != nullptr) {
+        std::unordered_set<std::string> originalOps;
+        for (auto&& node : function->get_ops()) {
+            if (!ngraph::op::is_parameter(node) &&
+                !ngraph::op::is_output(node) &&
+                !ngraph::op::is_constant(node)) {
+                originalOps.emplace(node->get_friendly_name());
+            }
+        }
 
-    const auto log = std::make_shared<Logger>(
-        "GraphCompiler",
-        parsedConfigCopy.logLevel(),
-        defaultOutput(parsedConfigCopy.compilerLogFilePath()));
+        auto clonedNetwork = CloneAndTransformNetwork(network);
+
+        std::unordered_set<std::string> supported;
+        std::unordered_set<std::string> unsupported;
+
+        std::unordered_set<std::string> splitNames;
+        std::unordered_set<std::string> concatNames;
+
+        std::vector<std::shared_ptr<ngraph::Node>> splits;
+        std::vector<std::shared_ptr<ngraph::Node>> concats;
+
+        for (CNNNetworkIterator itLayer{clonedNetwork.get()};
+             itLayer != CNNNetworkIterator();
+             itLayer++) {
+            auto layerIsSupported = [&] {
```

Comment:
1. `isLayerSupported` would be more grammatically correct, I guess
2. `&` capture not a good practice. Could you implement this lamba outside `for` loop with `std::shared_ptr<Node>` argument and explicitly declare capture list

Reply:
Done